service/src/main/java/bio/terra/axonserver/service/cromwellworkflow/CromwellWorkflowService.java

```diff
@@ -235,19 +238,20 @@ public CromwellApiWorkflowIdAndStatus submitWorkflow(
         out.write(mapper.writeValueAsString(workflowOptions).getBytes(StandardCharsets.UTF_8));
       }
 
-      // Adjoin the workspace-id label to the workflow.
       labels.put(WORKSPACE_ID_LABEL_KEY, workspaceId.toString());
-      try (OutputStream out = new FileOutputStream(tempLabelsFile.getFile())) {
-        out.write(mapper.writeValueAsString(labels).getBytes(StandardCharsets.UTF_8));
-      }
+      labels.put(USER_EMAIL_LABEL_KEY, samService.getUserStatusInfo(token).getUserEmail());
       if (workflowGcsUri != null) {
+        labels.put(GCS_SOURCE_LABEL_KEY, workflowGcsUri);
         InputStream inputStream =
             fileService.getFile(token, workspaceId, workflowGcsUri, /*convertTo=*/ null);
         Files.copy(
             inputStream,
             tempWorkflowSourceFile.getFile().toPath(),
             StandardCopyOption.REPLACE_EXISTING);
       }
+      try (OutputStream out = new FileOutputStream(tempLabelsFile.getFile())) {
+        out.write(mapper.writeValueAsString(labels).getBytes(StandardCharsets.UTF_8));
```

Comment:
this looks like users could replace our internal labels, which would be bad?  should we write our own ones last so they take priority?

Reply:
thats how its written right now - you can see above in the code that we overwrite any keys that might already exist for the labels we care about.